c/types.go

```diff
@@ -81,3 +86,20 @@ type Child struct {
 	cancel      func()
 	wait        func(Shutdown) error
 }
+
+// ChildNotification reports when a child has finished
```

Comment:
And it may finish without error, so error may be `nil`, right?

Reply:
Correct, I'll add that to the comment